src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleCache.java

```diff
@@ -0,0 +1,86 @@
+package net.sf.jabref.logic.citationstyle;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.sf.jabref.BibDatabaseContext;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.event.EntryChangedEvent;
+import net.sf.jabref.model.event.EntryRemovedEvent;
+
+import com.google.common.eventbus.Subscribe;
+
+
+/**
+ * Caches the generated Citations for quicker access
+ */
+public class CitationStyleCache {
```

Comment:
The library generates the citation with JavaScript. In the beginning it will take about a second, after the JavaScript engine warmup it is much quicker but still noticeable (100ms-200ms), especially if you scroll with the arrow keys over multiple entries.